### PR TITLE
Remove unnecessary properties from carb structures

### PR DESCRIPTION
**Pull request by darinkrauss**
- https://tidepool.atlassian.net/browse/LOOP-1417
- Remove unused uploadState from CachedCarbObject and DeletedCarbObject
- Remove unused isUploaded from StoredCarbEntry and DeletedCarbEntry
- Remove unused createdByCurrentApp, externalID, syncIdentifier, and isUploaded from NewCarbEntry

### CODE REVIEW

**Comment by darinkrauss** [comment]
Build: https://github.com/tidepool-org/LoopWorkspace/pull/211

File: Loop/Extensions/CarbStore+SimulatedCoreData.swift [hunk 1]
```diff
@@ -117,15 +117,13 @@ fileprivate extension StoredCarbEntry {
                                foodType: "Simulated",
                                absorptionTime: absorptionTime,
                                createdByCurrentApp: true,
-                               externalID: UUID().uuidString,
-                               isUploaded: false)
+                               externalID: UUID().uuidString)
```
Comment (nhamming) [inline Loop/Extensions/CarbStore+SimulatedCoreData.swift:120]: Just want to confirm that keeping this `externalID` was intended.

**Review by nhamming — APPROVED** [review]
LGTM

File: Loop/Extensions/CarbStore+SimulatedCoreData.swift (hunk 1 above)
Comment (darinkrauss) [reply Loop/Extensions/CarbStore+SimulatedCoreData.swift:120]: Yes, `externalID` is needed for the DIY user that transitions from pre-RDS to post-RDS DIY Loop. Pre-RDS, the `externalID` field is used by Nightscout to "remember" the Nightscout entry ID (used for later updates/deletes). The post-RDS Nightscout code will use this field (if it exists) just for updates/deletes of previously existing data, but it will no longer store an externalID into that field (and, in fact, it cannot) for new data.

In short, it exists for backwards compatibility for DIY Loop users for migration purposes.

File: Loop/View Controllers/CarbEntryViewController.swift [hunk 2]
```diff
@@ -104,8 +104,7 @@ final class CarbEntryViewController: LoopChartsTableViewController, Identifiable
                 quantity: quantity,
                 startDate: date,
                 foodType: foodType,
-                absorptionTime: absorptionTime,
-                externalID: originalCarbEntry?.externalID
```
Comment (ps2) [inline Loop/View Controllers/CarbEntryViewController.swift:108]: Isn't externalID how CarbStore knows it's an update, and not a new entry?

**Review by ps2 — APPROVED** [review]
LGTM, assuming the answer to my question is "no".

File: Loop/View Controllers/CarbEntryViewController.swift (hunk 2 above)
Comment (darinkrauss) [reply Loop/View Controllers/CarbEntryViewController.swift:108]: No. The fact that there is an `originalCarbEntry` means that it is an update, by definition. The existence of `originalCarbEntry` eventually makes it through the workflow to `LoopDataManager.addCarbEntry` which then does the "right thing". If an `originalCarbEntry` exists, then it calls `CarbStore.replaceCarbEntry` and, if not, then `CarbStore.addCarbEntry`.
